### ISSUE TITLE
Tag Aliasing - Project 250

### ISSUE THREAD

**Issue by leberrigan** [MotusDev/Motus-TO-DO#755]
I believe I've found a new case of tag aliasing where a new burst interval has emerged as a result of several _similar_ tags overlapping. This can be seen in [this plot](https://motus.org/data/receiver/deployment/plot/SG%2D6F19RPI385F8-2020%20Sask%20Passerine%2C%20Site2%2DWestman%28E%29.png?id=6044&n=100&firstTag=0).

This station is deployed in Saskatchewan, far from the summer range of Ipswich Sparrows and Tricoloured Blackbirds. While there are a few aliased tags (43684, 43254, 35991) that have the same burst interval as real deployed tags (**13.1 seconds**), there is one Ipswich Sparrow tag that doesn't (33052; **19.7 seconds**).

This aliased tag shares the same ID as a real tag in the area (484), and all the other tags real tags are also in the ~450-490 range. I'm not certain how this kind of aliasing could occur, but I can think of two possibilities that aren't mutually exclusive:
 1. It has something to do with the ID of the tags being associated with larger pulse intervals, which increases the chance of an overlap between two tags. 
 2. It turns out that the burst interval of the real tags (13.1 seconds) is close to two-thirds of the BI of the false tag (19.7/13.1 = 1.5038). That means it may be possible for a false detection to occur for every second detection of the real tag

![project250_tagAliasing](https://user-images.githubusercontent.com/35931796/86595816-aba1bc80-bf6f-11ea-9304-d21916bae06d.jpg)

**Comment by denislepage** [comment]
0.1   seconds difference should be more than the slop allowed, but in practice, the tag intervals are rarely exactly as advertised, so they could be closer in reality. Do you have the actual period measured during registration?

Your issue raises a good point though that period could potentially be aliased if you have enough tags with the same code but different intervals, even if the burst don’t overlap.   Two tags with the same number can lead to a run of 2 simply by bursting at an interval that matches another real period that tag finder is monitoring.

f you have 2 tags with the same number but different periods, at a 0.1 second precision (say 2.9 and 3.1 seconds), they may combine in ways that create other intervals. Say at 8.7 seconds for tag 1 (2.9 * 3), and 3.7 seconds later at second 12.4, you have the second tag bursting (3.1 * 4), this potentially produces a spurious run of 2 for a period that does exist elsewhere. If tags 1 and 2 are locked in existing runs, these combination should possibly be ignored, but I don’t know that for a fact. The other thing is that we are in fact using a higher precision than 0.1, making it that much less likely that 2 bursts will match a known tag, but it will occasionally happen. We should figure out how often, and whether it’s something worth worrying about.

If you multiple the number of tags sharing the same code number, then this can potentially escalate, something we may also want to do with a simulation. (start with 2 tags with the same number but different period, that run concurrently for a longish period of 5-10 minutes, and see how often that leads to spurious detections of other tags, and then gradually increase the number of tags. You could also try to run then for shorter periods, in a way that is more realistic, as those may be more likely to lead to spurious combinations than long running chains. If you start them at a random point rather than force them to start both at zero, you will more likely quickly run into this, but perhaps it doesn’t really matter. To be effective, our simulations should probably be based on realistic periods and run lengths.

Anyway, just some follow up ideas for another type of aliasing that we should try to understand.

Copying Brandon for his info.

**Comment by leberrigan** [comment]
This is a case where a single tag is appearing as if it were another. There are no two tags with the same ID that have been deployed here so it can't be the example case we spoke about in last weeks' call. The fact the aliased tag has the same ID as a real tag caught my attention. It's not clear whether the other aliased tags, which have different IDs but the same BI, are present for a similar reason as we believe those are only caused by multiple real tags with the same BI that have overlapping bursts. I think this may be related, but it will take some explaining. 

The tags in question have the following burst intervals:
 - 33032 (FALSE): 19.69729995727539
 - 45321 (TRUE): 13.09939956665039 

If we do the math, we should expect a large burst slop like Denis mentioned:
 ( 2 * **19.69729995727539** ) - ( **13.09939956665039** * 3 ) = **0.09640121459 seconds**

Accounting for the standard deviation of the burst interval makes little difference.

If we pretend there are a bunch of skipped bursts, we should expect this to increase significantly:
 ( 8 * **19.69729995727539** ) - ( **13.09939956665039** * 12 ) = **0.38560485839**

According to the [tag finder parameters](https://github.com/MotusDev/Motus-TO-DO/issues/465), it should not be identifying tags that have a burst slop greater than 0.001 seconds (10 milliseconds). While it does allow for this slop to increase if any bursts are skipped (up to 60), it only allows for the slop to 'expand' by _1 millisecond for each skipped burst_. That means for the first and second examples above we'd only expect burst slops of ~0.011 and ~0.017 seconds respectively.

So again I'm left scratching my head!

**Comment by leberrigan** [comment]
What if we're getting something strange where it's a combination of real detections of tag [45321](https://motus.org/data/tags/edit?projectID=250&id=45321) and aliased detections as a result of overlapping bursts, similar to the other aliased tags that recorded at that station.
![tag aliasing 3](https://user-images.githubusercontent.com/35931796/86980386-10206f80-c15a-11ea-88c8-41ae49aa1f22.png)

**Comment by denislepage** [comment]
This was also what I saw earlier in the examples I looked at. The original tags were still being picked up at the same time as the potential aliases, if this is what you mean in your plot. I thought tag finder would prevent this (i.e. the 4 pulses could only be used by a single run), but there may be some nuance allowing this (e.g. if this applies only to the first pulse of a burst, the subsequent pulses 2-4 could be combined in different ways with a second tag to create an alias).

Hopefully, we can learn from what we’ve asked Brandon to work on.
